pallets/verifiers/src/lib.rs

```diff
@@ -266,17 +309,29 @@ pub mod pallet {
 
         /// Register a new verification key.
         /// On success emit a `VkRegistered` event that contain the hash to use on `submit_proof`.
+        /// Lock some funds, which can be unlocked by calling `unregister_vk`.
         #[pallet::call_index(1)]
         #[pallet::weight(T::WeightInfo::register_vk(vk))]
-        pub fn register_vk(_origin: OriginFor<T>, vk: Box<I::Vk>) -> DispatchResultWithPostInfo {
+        pub fn register_vk(origin: OriginFor<T>, vk: Box<I::Vk>) -> DispatchResultWithPostInfo {
             log::trace!("Register vk");
             ensure!(
                 !Self::disabled().unwrap_or_default(),
                 on_disable_error::<T, I>()
             );
-            I::validate_vk(&vk).map_err(Error::<T, I>::from)?;
+            let account_id = ensure_signed(origin)?;
             let hash = I::vk_hash(&vk);
-            Vks::<T, I>::insert(hash, vk);
+            ensure!(
+                !Tickets::<T, I>::contains_key((&account_id, hash)),
+                Error::<T, I>::VerificationKeyAlreadyRegistered
+            );
+            I::validate_vk(&vk).map_err(Error::<T, I>::from)?;
+            let footprint = Footprint::from_encodable(&vk);
+            let ticket = T::Ticket::new(&account_id, footprint)?;
+            Tickets::<T, I>::insert((account_id, hash), ticket);
```

Comment:
So, we hold the currency like everyone register his own key and we ignore the reference count... Fair